src/eos-updater-poll-common.c

```diff
@@ -386,41 +391,37 @@ get_refspec_to_upgrade_on (gchar               **refspec_to_upgrade_on,
   if (booted_deployment == NULL)
     return FALSE;
 
-  if (!get_refspec_to_upgrade_on_from_deployment (sysroot,
-                                                  booted_deployment,
-                                                  &refspec,
-                                                  error))
+  if (!get_booted_refspec (booted_deployment,
+                           &refspec,
+                           &remote,
+                           &ref,
+                           &collection_ref,
+                           error))
```

Comment:
I think these can be renamed to `booted_refspec` `booted_remote`, `booted_ref`, `booted_collection_ref` etc. Then below in the branch where we check for the checkpoint, we can set the outparams to the overridden refspec, remote, and booted ref. If that branch isn't followed, then set the outparams to the `booted_refspec` and friends. That should obviate the need for the free/replace dance below.

Reply:
Makes sense. Keeping all the out params common is what pushed me what I had. Let me try reworking with them branched where necessary.